smac/intensification/parallel_successive_halving.py

```diff
@@ -0,0 +1,382 @@
+            return self.sh_instances[len(self.sh_instances) - 1].get_next_run(
+                challengers=challengers,
+                incumbent=incumbent,
+                chooser=chooser,
+                run_history=run_history,
+                repeat_configs=repeat_configs,
+            )
+
+        # If got to this point, there is no new SH addition viable
+        # not a SH has something to launch, so we have to wait
+        return RunInfoIntent.WAIT, RunInfo(
+            config=None,
+            instance="0",
+            instance_specific="0",
+            seed=0,
+            cutoff=self.cutoff,
+            capped=False,
+            budget=0.0,
+        )
+
+    def _add_new_SH(self, num_workers: int) -> bool:
+        """
+        Decides if it is possible to add a new SH, and adds it.
+        If a new SH is added, True is returned, else False.
+
+        Parameters:
+        -----------
+        num_workers: int
+            the maximum number of workers available
+            at a given time.
+
+        Returns
+        -------
+            Whether or not a successive halving instance was added
+        """
+        if len(self.sh_instances) >= num_workers:
+            return False
+
+        self.sh_instances[len(self.sh_instances)] = SuccessiveHalving(
+            stats=self.stats,
+            traj_logger=self.traj_logger,
+            rng=self.rs,
+            instances=self._instances,
+            instance_specifics=self._instance_specifics,
+            cutoff=self.cutoff,
+            deterministic=self.deterministic,
+            initial_budget=self.initial_budget,
+            max_budget=self.max_budget,
+            eta=self.eta,
+            num_initial_challengers=self.num_initial_challengers,
+            run_obj_time=self.run_obj_time,
+            n_seeds=self.n_seeds,
+            instance_order=self.instance_order,
+            adaptive_capping_slackfactor=self.adaptive_capping_slackfactor,
+            inst_seed_pairs=self.inst_seed_pairs,
+            min_chall=self.min_chall,
+            incumbent_selection=self.incumbent_selection,
+            identifier=len(self.sh_instances),
+        )
+
+        return True
+
+    def _sh_scheduling(self, strategy: str = 'more_advanced_iteration') -> typing.List[int]:
+        """
+        This procedure dictates what SH to prioritize in
+        launching jobs.
+
+        Parameters
+        ----------
+            strategy: str
+                What type of scheduling to follow
+
+        Returns
+        -------
+            List:
+                The order in which to query for new jobs
+
+        """
+
+        # This function might be called when no SuccessiveHalving instances
+        # exist (first iteration), so we return an empty list in that case
+        if len(self.sh_instances) == 0:
+            return []
+
+        expected = self.last_active_instance + 1
+        if strategy == 'ordinal' or expected == len(self.sh_instances):
```

Comment:
I'm actually wondering if we still need multiple strategies here? What would be the reason to not use 'more_advanced_iteration'?

Reply:
thanks, I have removed this.